utility/repoutil.py

```diff
@@ -92,14 +92,18 @@ def get_commit_hash(self, path=None, branch='master'):
                     repo_temp.clone()
                     break
         if branch == 'master':
+            if path is not None:
+                repo_temp.remove()
             return repo_temp.repo.head.commit.hexsha
         else:
             refs = repo_temp.repo.refs
             for ref in refs:
                 if ref.name == branch or ref.name == 'origin/{}'.format(branch):
-                    return ref.commit.hexsha
-        if path is not None:
-            repo_temp.remove()
+                    foundhash = ref.commit.hexsha
+            if foundhash is not None:
```

Comment:
found_hash must be initialized to None otherwise it will  say
`NameError: name 'found_hash' is not defined` if it will not be found in between refs
It can be set to None at the beginning of else statement on line 98
also please divide the name with _ so it is found_hash instead of foundhash

Reply:
I fixed that and also refactored all code.